In [1]:
import numpy as np
import pandas as pd

In [2]:
d1_d2 = pd.read_csv("D:/NYU_MS/Sem4/DL_in_Med/project/data/TADPOLE_D1_D2.csv",encoding = "utf-8")
d1_d2.head()

C:\Users\poorn\Anaconda3\lib\site-packages\IPython\core\interactiveshell.py:3058: DtypeWarning: Columns (471,473,474,487,488,489,490,491,492,493,494,495,496,497,498,499,500,501,502,503,504,505,506,507,508,509,510,511,512,513,514,515,516,517,518,519,520,521,522,523,524,525,526,527,528,529,530,531,532,533,534,535,536,537,538,539,540,541,542,543,544,545,546,547,548,549,550,551,552,553,554,555,556,557,558,559,560,561,562,563,569,570,572,573,574,575,576,577,578,579,580,581,582,583,584,585,586,587,588,589,590,591,592,593,594,595,596,597,599,601,606,607,608,609,610,611,612,613,614,615,616,617,618,619,620,621,624,625,626,627,628,629,630,631,632,633,634,636,637,638,639,640,641,642,643,644,645,646,647,648,649,650,651,652,653,654,655,656,657,658,659,660,661,663,664,665,666,667,668,669,670,671,672,673,674,675,676,677,678,679,680,681,682,683,684,685,686,687,688,689,690,691,692,693,694,695,696,697,698,699,700,701,702,703,704,705,706,707,708,709,710,711,712,713,714,715,716,717,718,719,720,721,722,723

RID        PTID VISCODE  SITE  D1  D2 COLPROT ORIGPROT    EXAMDATE DX_bl  \
0    2  011_S_0002      bl    11   1   1   ADNI1    ADNI1  2005-09-08    CN   
1    3  011_S_0003      bl    11   1   0   ADNI1    ADNI1  2005-09-12    AD   
2    3  011_S_0003     m06    11   1   0   ADNI1    ADNI1  2006-03-13    AD   
3    3  011_S_0003     m12    11   1   0   ADNI1    ADNI1  2006-09-12    AD   
4    3  011_S_0003     m24    11   1   0   ADNI1    ADNI1  2007-09-12    AD   

   ...  PHASE_UPENNBIOMK9_04_19_17  BATCH_UPENNBIOMK9_04_19_17  \
0  ...                                                           
1  ...                       ADNI1                 UPENNBIOMK9   
2  ...                                                           
3  ...                       ADNI1                 UPENNBIOMK9   
4  ...                                                           

  KIT_UPENNBIOMK9_04_19_17  STDS_UPENNBIOMK9_04_19_17  \
0                                                       
1            P06-MP02-MP01            P06-MP02-MP01/2   
2                                                       
3            P06-MP02-MP01            P06-MP02-MP01/2   
4                                                       

  RUNDATE_UPENNBIOMK9_04_19_17 ABETA_UPENNBIOMK9_04_19_17  \
0                                                           
1                   2016-12-14                      741.5   
2                                                           
3                   2016-12-14                      601.4   
4                                                           

  TAU_UPENNBIOMK9_04_19_17  PTAU_UPENNBIOMK9_04_19_17  \
0                                                       
1                    239.7                      22.83   
2                                                       
3                    251.7                      24.18   
4                                                       

   COMMENT_UPENNBIOMK9_04_19_17  update_stamp_UPENNBIOMK9_04_19_17  
0                                                                   
1                           NaN              2017-04-20 14:39:54.0  
2                                                                   
3                           NaN              2017-04-20 14:39:54.0  
4                                                                   

[5 rows x 1907 columns]

In [4]:
#no. of unique patients
len(d1_d2.RID.unique())

1737

In [5]:
#no. of different visits possible
len(d1_d2.VISCODE.unique())

22

In [6]:
df = d1_d2[['CDRSB','ADAS11','MMSE','RAVLT_immediate','Hippocampus','WholeBrain','Entorhinal','MidTemp','FDG','AV45','ABETA_UPENNBIOMK9_04_19_17','TAU_UPENNBIOMK9_04_19_17','PTAU_UPENNBIOMK9_04_19_17','APOE4','AGE']]
df.head()

CDRSB  ADAS11  MMSE  RAVLT_immediate  Hippocampus  WholeBrain  Entorhinal  \
0    0.0   10.67  28.0             44.0       8336.0   1229740.0      4177.0   
1    4.5   22.00  20.0             22.0       5319.0   1129830.0      1791.0   
2    6.0   19.00  24.0             19.0       5446.0   1100060.0      2427.0   
3    3.5   24.00  17.0             31.0       5157.0   1095640.0      1596.0   
4    8.0   25.67  19.0             23.0       5139.0   1088560.0      1175.0   

   MidTemp      FDG  AV45 ABETA_UPENNBIOMK9_04_19_17 TAU_UPENNBIOMK9_04_19_17  \
0  27936.0  1.36926   NaN                                                       
1  18422.0  1.09079   NaN                      741.5                    239.7   
2  16972.0  1.06360   NaN                                                       
3  17330.0  1.10384   NaN                      601.4                    251.7   
4  16398.0  1.03871   NaN                                                       

  PTAU_UPENNBIOMK9_04_19_17  APOE4   AGE  
0                              0.0  74.3  
1                     22.83    1.0  81.3  
2                              1.0  81.3  
3                     24.18    1.0  81.3  
4                              1.0  81.3

In [7]:
df.describe()

CDRSB       ADAS11         MMSE  RAVLT_immediate   Hippocampus  \
count  8963.000000  8910.000000  8932.000000      8831.000000   6802.000000   
mean      2.171427    11.394677    26.592253        34.480353   6681.723758   
std       2.810109     8.632739     3.948484        13.611643   1239.952328   
min       0.000000     0.000000     0.000000         0.000000   2219.000000   
25%       0.000000     5.330000    25.000000        25.000000   5830.000000   
50%       1.000000     9.000000    28.000000        33.000000   6759.500000   
75%       3.000000    15.000000    29.000000        44.000000   7583.000000   
max      18.000000    70.000000    30.000000        75.000000  11207.000000   

         WholeBrain   Entorhinal       MidTemp          FDG         AV45  \
count  7.689000e+03  6469.000000   6469.000000  3352.000000  2118.000000   
mean   1.011816e+06  3444.718813  19203.260318     1.208239     1.194780   
std    1.119606e+05   811.771966   3138.155124     0.160831     0.227518   
min    6.490910e+05  1041.000000   8044.000000     0.636804     0.835796   
25%    9.337950e+05  2905.000000  17183.000000     1.109722     1.010035   
50%    1.009330e+06  3482.000000  19240.000000     1.219805     1.113975   
75%    1.088250e+06  4013.000000  21397.000000     1.314253     1.362275   
max    1.486040e+06  6711.000000  32189.000000     1.753320     2.669210   

              APOE4           AGE  
count  12729.000000  12741.000000  
mean       0.537277     73.814865  
std        0.655835      6.973068  
min        0.000000     54.400000  
25%        0.000000     69.600000  
50%        0.000000     73.800000  
75%        1.000000     78.700000  
max        2.000000     91.400000

In [8]:
df.isna().sum()

CDRSB                          3778
ADAS11                         3831
MMSE                           3809
RAVLT_immediate                3910
Hippocampus                    5939
WholeBrain                     5052
Entorhinal                     6272
MidTemp                        6272
FDG                            9389
AV45                          10623
ABETA_UPENNBIOMK9_04_19_17        2
TAU_UPENNBIOMK9_04_19_17          2
PTAU_UPENNBIOMK9_04_19_17         3
APOE4                            12
AGE                               0
dtype: int64

In [11]:
new_df = df.dropna()
new_df.describe()

CDRSB       ADAS11         MMSE  RAVLT_immediate   Hippocampus  \
count  1121.000000  1121.000000  1121.000000      1121.000000   1121.000000   
mean      1.495094     9.457627    27.561106        38.431757   6991.737734   
std       1.935967     7.230654     2.961066        13.319127   1150.571690   
min       0.000000     0.000000     9.000000         0.000000   3290.000000   
25%       0.000000     5.000000    26.000000        28.000000   6253.000000   
50%       1.000000     8.000000    29.000000        38.000000   7108.000000   
75%       2.000000    11.000000    30.000000        48.000000   7765.000000   
max      14.000000    58.000000    30.000000        73.000000  11207.000000   

         WholeBrain   Entorhinal       MidTemp          FDG         AV45  \
count  1.121000e+03  1121.000000   1121.000000  1121.000000  1121.000000   
mean   1.040881e+06  3582.144514  19899.179304     1.253554     1.200493   
std    1.082994e+05   753.820786   2887.020971     0.153618     0.229061   
min    7.152520e+05  1082.000000   9791.000000     0.697264     0.838537   
25%    9.686460e+05  3127.000000  18079.000000     1.162600     1.015720   
50%    1.038000e+06  3637.000000  19859.000000     1.270440     1.112820   
75%    1.108050e+06  4104.000000  21885.000000     1.357170     1.379290   
max    1.486040e+06  5896.000000  32189.000000     1.707170     2.669210   

             APOE4          AGE  
count  1121.000000  1121.000000  
mean      0.504906    72.047725  
std       0.642417     6.911212  
min       0.000000    55.000000  
25%       0.000000    67.600000  
50%       0.000000    72.100000  
75%       1.000000    76.600000  
max       2.000000    90.300000

No. of rows with no missing records is 1121!

In [12]:
new_df.isna().sum()

CDRSB                         0
ADAS11                        0
MMSE                          0
RAVLT_immediate               0
Hippocampus                   0
WholeBrain                    0
Entorhinal                    0
MidTemp                       0
FDG                           0
AV45                          0
ABETA_UPENNBIOMK9_04_19_17    0
TAU_UPENNBIOMK9_04_19_17      0
PTAU_UPENNBIOMK9_04_19_17     0
APOE4                         0
AGE                           0
dtype: int64